homeassistant/components/template/trigger_entity.py

```diff
@@ -47,22 +52,47 @@ def referenced_blueprint(self) -> str | None:
         """Return referenced blueprint or None."""
         return self.coordinator.referenced_blueprint
 
+    @property
+    def available(self) -> bool:
+        """Return availability of the entity."""
+        if self._state_render_error:
+            return False
+
+        return super().available
+
     @callback
     def _render_script_variables(self) -> dict:
         """Render configured variables."""
         return self.coordinator.data["run_variables"]
 
+    def _render_templates(self, variables: dict[str, Any]) -> None:
+        """Render templates."""
+        self._state_render_error = False
+        rendered = dict(self._static_rendered)
+
+        # If state fails to render, the entity should go unavailable.  Render the
+        # state as a simple template because the result should always be a string or None.
```

Comment:
```suggestion
        # If state fails to render, the entity should go unavailable. Render the
        # state as a simple template because the result should always be a string or None.
```